pkg/controller/nodelifecycle/scheduler/taint_manager.go

```diff
@@ -457,14 +448,6 @@ func (tc *NoExecuteTaintManager) handleNodeUpdate(ctx context.Context, nodeUpdat
 	if len(pods) == 0 {
 		return
 	}
-	// Short circuit, to make this controller a bit faster.
```

Comment:
Remove this snippets for we have another loop follow up and we will also verify the taints length in `processPodOnNode`, I thinks it has no performance improvement.